std/js/lib/intl/DateTimeFormat.hx

```diff
@@ -0,0 +1,392 @@
+	@:pure function formatToParts(date:Date):Array<DateTimeFormatPart>; 
+
+	/**
+		Returns a new object with properties reflecting the locale and formatting options
+		computed during initialization of the object.
+	**/
+	@:pure function resolvedOptions():DateTimeFormatResolvedOptions;
+
+	/**
+		Returns an array containing those of the provided locales that are supported
+		without having to fall back to the runtime's default locale.
+	**/
+	@:overload(function(locales:Array<String>, ?options:DateTimeFormatSupportedLocalesOfOptions):Array<String> {})
+	@:pure static function supportedLocalesOf(locales:String, ?options:DateTimeFormatSupportedLocalesOfOptions):Array<String>;
+}
+
+typedef DateTimeFormatOptions = {
+	/**
+		The locale matching algorithm to use.
+		The default is `BestFit`.
+	**/
+	var ?localeMatcher:LocaleMatcher;
+
+	/**
+		The time zone to use. The only value implementations must recognize is `"UTC"`;
+		the default is the runtime's default time zone. Implementations may also recognize
+		the time zone names of the [IANA time zone database](https://www.iana.org/time-zones),
+		such as `"Asia/Shanghai"`, `"Asia/Kolkata"`, `"America/New_York"`.
+	**/
+	var ?timeZone:String;
+
+	/**
+		Whether to use 12-hour time (as opposed to 24-hour time).
+		The default is locale dependent.
+		This option overrides the hc language tag and/or the `hourCycle` option in case both are present.
+	**/
+	var ?hour12:Bool;
+
+	/**
+		The hour cycle to use. This option overrides the `hc` language tag, if both are present,
+		and the `Hour12` option takes precedence in case both options have been specified.
+	**/
+	var ?hourCycle:HourCycle;
+
+	/**
+		The format matching algorithm to use.
+		The default is `BestFit`.
+		See the following paragraphs for information about the use of this property.
+	**/
+	var ?formatMatcher:FormatMatcher;
+
+	/**
+		The representation of the weekday.
+	**/
+	var ?weekday:WeekdayRepresentation;
+
+	/**
+		The representation of the era.
+	**/
+	var ?era:EraRepresentation;
+
+	/**
+		The representation of the year.
+	**/
+	var ?year:YearRepresentation;
+
+	/**
+		The representation of the month.
+	**/
+	var ?month:MonthRepresentation;
+
+	/**
+		The representation of the day.
+	**/
+	var ?day:DayRepresentation;
+
+	/**
+		The representation of the hour. 
+	**/
+	var ?hour:HourRepresentation;
+
+	/**
+		The representation of the minute.
+	**/
+	var ?minute:MinuteRepresentation;
+
+	/**
+		The representation of the second.
+	**/
+	var ?second:SecondRepresentation;
+
+	/**
+		The representation of the time zone name. 
+	**/
+	var ?timeZoneName:TimeZoneName;
+}
+
+typedef DateTimeFormatResolvedOptions = {
+	/**
+		The BCP 47 language tag for the locale actually used.
+		If any Unicode extension values were requested in the input BCP 47 language tag that led to this locale,
+		the key-value pairs that were requested and are supported for this locale are included in `locale`.
+	**/
+	var locale(default,never):String;
```

Comment:
shouldn't these be `final`?

Reply:
Can I use `final`? There are yet many `(default, never)` style codes in other places.

Reply:
Codebases like hxnodejs still use the latter for backwards compatibility with Haxe 3.4. The std lib doesn't need to be backwards-compatible with older Haxe versions though.

Reply:
Thanks. I'll change these to 'final'.